File: src/libs/ThirdParty.VBSuppliers/Properties/JonView/Models/JonViewSearchResponse.vb
Imports System.Xml.Serialization

<Serializable(), XmlRoot("message")>
Public Class JonViewSearchResponse
    <XmlElement("alternatelistseg")> Public Property Response As ResponseDetails

    <Serializable(), XmlType("alternatelistseg")>
    Public Class ResponseDetails
        <XmlElement("listrecord")> Public Property Rooms As New List(Of Room)
    End Class

    <Serializable()>
    Public Class Room
        Public Property prodcode As String
        Public Property status As String
        Public Property productname As String
        Public Property currencycode As String
        Public Property dayprice As String
        Public Property suppliercode As String
        <XmlElement("productnamedetails")> Public Property roomDetails As New Details
        <XmlElement("cancellationpolicy")> Public Property cancellationPolicy As CancellationPolicy
    End Class

    <Serializable()>
    Public Class Details
        Public Property board As String
        Public Property roomtype As String
    End Class

    Public Class CancellationPolicy
        <XmlElement("canpolicyitem")> Public Property item As New CancellationDetails
    End Class

    Public Class CancellationDetails
        Public Property fromdays As String
    End Class

End Class

File: src/libs/iVectorOne.Suppliers/JonView/Models/JonViewSearchResponse.cs
namespace iVectorOne.Suppliers.JonView
{
    using System;
    using System.Collections.Generic;
    using System.Xml.Serialization;

    [Serializable()]
    [XmlRoot("message")]
    public class JonViewSearchResponse
    {
        [XmlElement("alternatelistseg")]
        public ResponseDetails Response { get; set; }

        [Serializable()]
        [XmlType("alternatelistseg")]
        public class ResponseDetails
        {
            [XmlElement("listrecord")]
            public List<Room> Rooms { get; set; } = new List<Room>();
        }

        [Serializable()]
        public class Room
        {
            public string prodcode { get; set; }
            public string status { get; set; }
            public string productname { get; set; }
            public string currencycode { get; set; }
            public string dayprice { get; set; }
            public string suppliercode { get; set; }
            [XmlElement("productnamedetails")]
            public Details roomDetails { get; set; } = new Details();
            [XmlElement("cancellationpolicy")]
            public CancellationPolicy cancellationPolicy { get; set; }
        }

        [Serializable()]
        public class Details
        {
            public string board { get; set; }
            public string roomtype { get; set; }
        }

        public class CancellationPolicy
        {
            [XmlElement("canpolicyitem")]
            public CancellationDetails item { get; set; } = new CancellationDetails();
        }

        public class CancellationDetails
        {
            public string fromdays { get; set; }
        }

    }
}